### PR TITLE
CHIA-907 Augment uncurry_puzzle to accept both types of programs

### PR DESCRIPTION
**Pull request by AmineKhaldi**
This allows us to use serialized programs without the need to convert them into programs.

### CODE REVIEW

**Review by Quexington — APPROVED** [review]

File: chia/wallet/uncurried_puzzle.py [hunk 1]
```diff
@@ -11,5 +13,5 @@ class UncurriedPuzzle:
     args: Program
 
 
-def uncurry_puzzle(puzzle: Program) -> UncurriedPuzzle:
+def uncurry_puzzle(puzzle: Union[Program, SerializedProgram]) -> UncurriedPuzzle:
```
Comment (arvidn) [inline chia/wallet/uncurried_puzzle.py:16]: if `puzzle` is a `SerializedProgram`, `uncurry()` returns a tuple of `SerializedProgram`. The member of `UncurriedPuzzle` is `Program`, `Program`. This doesn't look right

**Review by arvidn — COMMENTED** [review]
it doesn't look right. The `uncurry()` member functions return different types.

File: chia/wallet/uncurried_puzzle.py (hunk 1 above)
Comment (AmineKhaldi) [reply chia/wallet/uncurried_puzzle.py:16]: If the puzzle is a `SerializedProgram`, `uncurry()` returns a tuple of `Program` as well.

Comment (arvidn) [reply chia/wallet/uncurried_puzzle.py:16]: not according to the type stub: https://github.com/Chia-Network/chia_rs/blob/main/wheel/python/chia_rs/chia_rs.pyi#L1976

Comment (arvidn) [reply chia/wallet/uncurried_puzzle.py:16]: oh, wait. `ChiaProgram` is an alias for the chia-blockchain python implementation

**Review by arvidn — APPROVED** [review]

**Pull request merged**